packages/webview_flutter/webview_flutter_android/lib/src/android_webview_controller.dart

```diff
@@ -780,36 +848,168 @@ class AndroidWebViewWidget extends PlatformWebViewWidget {
+    this.displayWithHybridComposition = false,
+    @visibleForTesting InstanceManager? instanceManager,
+    @visibleForTesting
+    this.platformViewsServiceProxy = const PlatformViewsServiceProxy(),
+  }) : instanceManager =
+            instanceManager ?? android_webview.JavaObject.globalInstanceManager;
+
+  /// The reference to the Android native view that should be shown.
+  final android_webview.View customView;
+
+  /// The [PlatformWebViewController] that allows controlling the native web
+  /// view.
+  final PlatformWebViewController controller;
+
+  /// The layout direction to use for the embedded WebView.
+  final TextDirection layoutDirection;
+
+  /// The `gestureRecognizers` specifies which gestures should be consumed by the
+  /// web view.
+  ///
+  /// It is possible for other gesture recognizers to be competing with the web
+  /// view on pointer events, e.g. if the web view is inside a [ListView] the
+  /// [ListView] will want to handle vertical drags. The web view will claim
+  /// gestures that are recognized by any of the recognizers on this list.
+  ///
+  /// When `gestureRecognizers` is empty (default), the web view will only handle
+  /// pointer events for gestures that were not claimed by any other gesture
+  /// recognizer.
+  final Set<Factory<OneSequenceGestureRecognizer>> gestureRecognizers;
+
+  /// Maintains instances used to communicate with the native objects they
+  /// represent.
+  ///
+  /// This field is exposed for testing purposes only and should not be used
+  /// outside of tests.
+  @visibleForTesting
+  final InstanceManager instanceManager;
+
+  /// Proxy that provides access to the platform views service.
+  ///
+  /// This service allows creating and controlling platform-specific views.
+  @visibleForTesting
+  final PlatformViewsServiceProxy platformViewsServiceProxy;
+
+  /// Whether the [WebView] will be displayed using the Hybrid Composition
+  /// PlatformView implementation.
+  ///
+  /// For most use cases, this flag should be set to false. Hybrid Composition
+  /// can have performance costs but doesn't have the limitation of rendering to
+  /// an Android SurfaceTexture. See
+  /// * https://flutter.dev/docs/development/platform-integration/platform-views#performance
+  /// * https://github.com/flutter/flutter/issues/104889
+  /// * https://github.com/flutter/flutter/issues/116954
+  ///
+  /// Defaults to false.
+  final bool displayWithHybridComposition;
+}
+
+/// Represents a Flutter implementation of the Android [View](https://developer.android.com/reference/android/view/View)
+/// that is created by the host platform when web content needs to be displayed
+/// in fullscreen mode.
+///
+/// The [AndroidCustomViewWidget] cannot be manually instantiated and is
+/// provided to the host application through the callbacks specified using the
+/// [AndroidWebViewController.setCustomViewCallbacks] method.
+class AndroidCustomViewWidget extends StatelessWidget {
```

Comment:
I think we would be better off making this class private or at least `@visibleForTesting`. We pass it as a parameter in the callback anyways so a user shouldn't need create an instance of this class anyways. In the `OnShowCustomViewCallback`, we can just use a `Widget` as the type.

Reply:
I have added the `@visibleForTesting` annotation here and also for the `AndroidCustomViewWidgetCreationParams`, also adding some additional explanation to the API docs.